arm/Microsoft.OperationalInsights/workspaces/dataSources/deploy.bicep

```diff
@@ -0,0 +1,81 @@
+@description('Required. Name of the Log Analytics workspace')
+param logAnalyticsWorkspaceName string
+
+@description('Required. Name of the solution')
+param name string
+
+@description('Required. The kind of the DataSource.')
+@allowed([
+  'AzureActivityLog'
+  'WindowsEvent'
+  'WindowsPerformanceCounter'
+  'IISLogs'
+  'LinuxSyslog'
+  'LinuxSyslogCollection'
+  'LinuxPerformanceObject'
+  'LinuxPerformanceCollection'
+])
+param kind string = 'AzureActivityLog'
+
+@description('Optional. Tags to configure in the resource.')
+param tags object = {}
+
+@description('Optional. Id of the resource to be linked.')
+param linkedResourceId string = ''
+
+@description('Optional. Windows event log name to configure when kind is WindowsEvent.')
+param eventLogName string = ''
+
+@description('Optional. Windows event types to configure when kind is WindowsEvent.')
+param eventTypes array = []
+
+@description('Optional. Name of the object to configure when kind is WindowsPerformanceCounter or LinuxPerformanceObject.')
+param objectName string = ''
+
+@description('Optional. Name of the instance to configure when kind is WindowsPerformanceCounter or LinuxPerformanceObject.')
+param instanceName string = '*'
+
+@description('Optional. Interval in seconds to configure when kind is WindowsPerformanceCounter or LinuxPerformanceObject.')
+param intervalSeconds int = 60
+
+@description('Optional. List of counters to configure when the kind is LinuxPerformanceObject.')
+param performanceCounters array = []
+
+@description('Optional. Counter name to configure when kind is WindowsPerformanceCounter.')
+param counterName string = ''
+
+@description('Optional. State to configure when kind is IISLogs or LinuxSyslogCollection or LinuxPerformanceCollection.')
+param state string = ''
+
+@description('Optional. System log to configure when kind is LinuxSyslog.')
+param syslogName string = ''
+
+@description('Optional. Severities to configure when kind is LinuxSyslog.')
+param syslogSeverities array = []
+
+@description('Optional. Customer Usage Attribution id (GUID). This GUID must be previously registered')
+param cuaId string = ''
+
+module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource symbolicname 'Microsoft.OperationalInsights/workspaces/dataSources@2020-08-01' = {
```

Comment:
Maybe something else than symbolic name? In case you need to reference we in the future

Reply:
good point, updated :)